File: src/main/scala/WordNet.scala
import argonaut._
import Argonaut._

case class WordNet(byId : Map[String, WordNetEntry])

case class WordNetEntry(parents : Seq[String], lemmas : Seq[String], defn : String, isInstance : Boolean)

object WordNetEntry {
  implicit def WordNetCodeJson : CodecJson[WordNetEntry] = 
    casecodec4(WordNetEntry.apply, WordNetEntry.unapply)("parents", "lemmas", "defn", "is_instance")
}

File: annotool-rs/src/wordnet.rs
use std::collections::HashMap;

#[derive(Debug,Serialize,Deserialize)]
pub struct WordNet(HashMap<String, WordNetEntry>);

#[derive(Debug,Serialize,Deserialize)]
pub struct WordNetEntry {
    parents : Vec<String>,
    lemmas : Vec<String>,
    defn : String,
    is_instance : bool
}

impl WordNet {
    pub fn lemmas(&self, id : &str) -> String {
        match self.0.get(id) {
            Some(wne) => {
                wne.lemmas.join(", ")
            },
            None =>
                "".to_string()
        }
    }

    pub fn defn(&self, id : &str) -> String {
        match self.0.get(id) {
            Some(wne) => wne.defn.clone(),
            None => "".to_string()
        }
    }
}
